src/input/composite_device/mod.rs

```diff
@@ -1803,6 +1830,28 @@ impl CompositeDevice {
         Ok(())
     }
 
+    // Deterimines if a given target device kind is already running
+    async fn target_kind_running(&self, mut kind: String) -> Result<bool, Box<dyn Error>> {
+        // TODO: Save this on the DS5 target device so we can properly look it up.
+        if kind == "ds5" {
+            kind = "ds5_edge".to_string();
+        }
```

Comment:
Instead of taking in a `mut String`, you could just pass a `&str` to avoid extra cloning. E.g.
```rust
let kind = match kind {
    "ds5" => "ds5_edge",
    _ => kind,
};
...
```